packages/core/parcel/src/bin.js

```diff
@@ -2,7 +2,10 @@
 
 'use strict';
 
-if (process.env.PARCEL_BUILD_ENV !== 'production') {
+if (
+  process.env.PARCEL_BUILD_ENV !== 'production' ||
+  process.env.PARCEL_SELF_BUILD
```

Comment:
This new environment variable is needed so we can replace PARCEL_BUILD_ENV while parcel is building itself...